ui/pages/confirmations/components/simulation-details/simulation-details.tsx

```diff
@@ -206,10 +197,18 @@ export const SimulationDetails: React.FC<SimulationDetailsProps> = ({
   setLoadingComplete();
 
   const { error } = simulationData;
+
+  if (
```

Comment:
I think it might make sense to move all the error handling inside the error component:
https://github.com/MetaMask/metamask-extension/pull/23726/files

Reply:
For the sake of the bug fix, ideally we'd do the minimum here hence why I've just changed the condition to rely on the error code as that could also be considered a bug.

If we move the rendering around in future, I'd still lean towards keeping this condition at the top level as it results in nothing being rendered, where as a component function implies something would be rendered within the component.

Reply:
> I'd still lean towards keeping this condition at the top level as it results in nothing being rendered

That's a very good point. Thanks for walking me through your thought process on that.